armi/settings/setting.py

```diff
@@ -110,7 +112,11 @@ def __init__(
             Whether this should be considered an "environment" setting. These can be
             used by the Case system to propagate environment options through
             command-line flags.
-
+        oldNames : list of tuple, optional
+            List of previous names that this setting used to have, along with optional
+            expiration dates. These can aid in automatic migration of old inputs. When
+            provided, if it is appears that the expiration date has passed, old names
+            will result in errors.
```

Comment:
maybe clarify this is expected only if an input file is loaded that has to be updated by the user to proceed with this version of the code.